driver-core/src/main/java/com/datastax/driver/core/Message.java

```diff
@@ -180,9 +180,21 @@ ByteBuffer pagingState() {
             }
         }
 
-        abstract Request copy();
+        Request copy() {
+            Request request = copyInternal();
+            request.setCustomPayload(this.getCustomPayload());
+            return request;
+        }
+
+        abstract Request copyInternal();
```

Comment:
I can't think of a better name either. If we make the methods protected it's acceptable.